frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -642,13 +680,41 @@ export default ({
           messageId: null
         })
       }
-    }, 500)
-  },
-  watch: {
-    currentChatRoomId (to, from) {
+    }, 500),
+    archiveMessageState (chatRoomId = null) {
+      if (!this.isJoinedChatRoom(chatRoomId)) {
+        return
+      }
+      const unit = this.chatRoomSettings?.actionsPerPage || CHATROOM_ACTIONS_PER_PAGE
+      const from = this.latestEvents.length ? GIMessage.deserialize(this.latestEvents[0]).hash() : null
+      const to = this.latestEvents.length
+        ? GIMessage.deserialize(this.latestEvents[this.latestEvents.length - 1]).hash()
+        : null
+
+      // NOTE: save messages in the browser storage, but not more than 3 pages of events
+      if (this.latestEvents.length >= 3 * unit) {
+        sbp('gi.db/archive/delete', this.getArchiveKeyFromChatRoomId(chatRoomId))
```

Comment:
The way these messages are deleted doesn't make much sense to me.

Instead of keeping the number of stored events limited to `3 * unit`, this just deletes all of these events. But when then will it save the events again?

 This just doesn't seem right to me.

Reply:
1. I need to define the constant value for the number 3. (Not sure 3 is the best choice. It could be 5 or any other)
2. Do you remember `MAX_ARCHIVED_PERIODS` which we use in group contract? We archive payments but no more than `MAX_ARCHIVED_PERIODS` periods of payments. Here, it works the same too. We are not going to save more than 3 pages of events for each channel. The events are always saved to the IndexedDB unless the loaded events are for more than 3 pages or user didn't load any events(old or new) after he opens the page.

Reply:
OK, I cannot follow everything you're doing here. However, there is an easy way to make sure everything works correctly: make it easy to detect bugs.

So, please for now set `CHATROOM_MAX_ARCHIVE_ACTION_PAGES` to `2`. This way, if there's a bug in the archiving code, we will catch it earlier. Then later we can set it back to `3` after we're sure that everything is working OK.

Reply:
Okay. I will set `CHATROOM_MAX_ARCHIVE_ACTION_PAGES` to `2` then.